primefaces/src/main/java/org/primefaces/convert/DateTimeConverter.java

```diff
@@ -25,16 +25,60 @@
 
 import java.text.DateFormat;
 import java.text.SimpleDateFormat;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
 import java.util.HashMap;
+import java.util.Locale;
 import java.util.Map;
+import javax.faces.component.UIComponent;
+import javax.faces.context.FacesContext;
 import javax.faces.convert.ConverterException;
+import org.primefaces.component.datepicker.DatePicker;
 import org.primefaces.util.CalendarUtils;
 import org.primefaces.util.HTML;
 
 public class DateTimeConverter extends javax.faces.convert.DateTimeConverter implements ClientConverter {
 
     private Map<String, Object> metadata;
 
+    @Override
+    public Object getAsObject(FacesContext context, UIComponent component, String value) {
+        if (component instanceof DatePicker) {
+            return getAsObject(context, (DatePicker) component, value);
+        }
+        return super.getAsObject(context, component, value);
+    }
+
+    public Object getAsObject(FacesContext context, DatePicker datePicker, String value) {
+        if (value == null) {
+            return null;
+        }
+        String type = getType();
+        boolean isDate = "date".equals(type);
+        boolean isLocalDateTime = "localDateTime".equals(type);
+        if ("range".equals(datePicker.getSelectionMode()) && (isDate || isLocalDateTime)) {
```

Comment:
We can probably remove the range check...

Reply:
Range and show time are two different things. But we should probably look at `showTime="false"` and else fallback to super.

Reply:
@jepsar No, they are not always different things. You can currently actually set `selectionMode="range"` and `showTime="true"` even if there is an issue (see #5037). And you can also use the `DatePicker` as filter with `type="date"` for `DateTimeConverter`, set `selectionMode="single"` and `showTime="true"`.

Reply:
+1 for `selectionMode="single"`. I'll check for that and add an IT.